### PR TITLE
Interface properties

### PR DESCRIPTION
**Pull request by chriskohlhoff**

### CODE REVIEW

File: wording.md [hunk 1]
```diff
@@ -507,877 +510,998 @@ The directionality properties conform to the following specification:
...
+| `bulk_then_t` | The executor type satisfies the `BulkThenExecutor` requirements. |
 
 `S::static_query_v<Executor>` is true if and only if `Executor` fulfills `S`'s requirements.
 
-The `oneway_t`, `twoway_t` and `then_t` properties are not mutually exclusive.
-
-##### `twoway_t` customization points
-
-In addition to conforming to the above specification, the `twoway_t` property provides the following customization:
+The `oneway_t`, `twoway_t`, `then_t`, `bulk_oneway_t`, `bulk_twoway_t` and `bulk_then_t` properties are mutually exclusive.
```
Comment (jaredhoberock) [inline wording.md]: @chriskohlhoff Just for my understanding, do the interface-changing property types enforce mutual exclusion similar to the behavioral properties?

File: wording.md [hunk 2]
```diff
@@ -507,877 +510,998 @@ The directionality properties conform to the following specification:
...
-  executor require(Property) const;
+*Remarks:* This function shall not participate in overload resolution unless `then_t::static_query_v<Executor>` is false and `bulk_then_t::static_query_v<Executor>` is true.
 
-  template <class Property>
-  typename Property::polymorphic_query_result_type query(Property) const;
+#### `then_t` polymorphic wrapper
 
-  template<class Function>
-    void execute(Function&& f) const;
+TODO
```
Comment (jaredhoberock) [inline wording.md]: @chriskohlhoff Re: TODO: Is there some question over what this wording should say?

File: wording.md [hunk 3]
```diff
@@ -507,877 +510,998 @@ The directionality properties conform to the following specification:
...
 *Remarks:* This function shall not participate in overload resolution unless:
-* `CONTAINS_PROPERTY(p, OtherSupportableProperties)` , where `p` is each property in `SupportableProperties...`.
+* `CONTAINS_PROPERTY(execution::twoway_t, SupportableProperties)`,
+* and `CONTAINS_PROPERTY(execution::bulk_t, SupportableProperties)`.
 
-*Effects:* `*this` targets a copy of `e` initialized with `std::move(e)`.
+*Effects:* Performs `e.bulk_twoway_execute(f2, n, rf2, sf2)`, where:
 
-```
-template<class... OtherSupportableProperties> executor(executor<OtherSupportablePro
```
Comment (jaredhoberock) [inline wording.md]: @chriskohlhoff Same question for this TODO.

**Comment by chriskohlhoff** [comment]
unlike the behavioural properties i dont think this is enforced, except
by the polymorphic wrapper.

**Comment by chriskohlhoff** [comment]
from memory then_execute and bulk_then_execute were missing from the
polymorphic wrapper in the previous revision. i put this in as a
placeholder since then_excute is up in the air.

**Comment by jaredhoberock** [comment]
@chriskohlhoff Thanks, makes sense to me.

**Comment by jaredhoberock** [comment]
Thanks Chris, merged.

**Pull request merged**